lib/opam.ml

```diff
@@ -149,27 +159,29 @@ module Package_summary = struct
     hashes : OpamHash.t list;
     dev_repo : string option;
     depexts : (OpamSysPkg.Set.t * OpamTypes.filter) list;
+    vendored : bool;
```

Comment:
I would kinda argue that it becomes package metadata, since you could see it as this:

```ocaml

type version_constraint = Vendored | With_test | ...

{
...
  additional_version_constraints: version_constraint list;
...
}
```

I can move this away somewhere but interpreting this module/type as "opam-monorepo's view of what this particular OPAM package is" seems to me quite sensible.

Reply:
I think it's unrelated to opam and is completely tied to our view and the result of the solver. The fact that you have to explicitly pass the `vendored` boolean whenever you're building a package summary indicates that it is oddly located to me.

Reply:
To clarify this module wraps up opam functionalities to have a simpler and more suited API for us so I do believe it's not the right place.

Package summary is used in `Duniverse` for instance, where the vendored status should not be handled because only vendored packages go there.